src/definitions.rs

```diff
@@ -0,0 +1,306 @@
+    #[cfg(not(feature = "no_module"))]
+    pub fn modules(&self) -> impl Iterator<Item = (String, String)> + '_ {
+        let mut m = self
+            .engine
+            .global_sub_modules
+            .iter()
+            .map(move |(name, module)| {
+                (
+                    name.to_string(),
+                    format!("module {name};\n\n{}", module.declaration(self)),
+                )
+            })
+            .collect::<Vec<_>>();
+
+        m.sort_by(|(name1, _), (name2, _)| name1.cmp(name2));
+
+        m.into_iter()
+    }
+}
+
+impl Module {
+    fn declaration(&self, decl: &Definitions) -> String {
+        let mut s = String::new();
+        self.write_declaration(&mut s, decl).unwrap();
+        s
+    }
+
+    fn write_declaration(&self, writer: &mut dyn fmt::Write, decl: &Definitions) -> fmt::Result {
+        let mut first = true;
+
+        let mut vars = self.iter_var().collect::<Vec<_>>();
+        vars.sort_by(|(a, _), (b, _)| a.cmp(b));
+
+        for (name, _) in vars {
+            if !first {
+                writer.write_str("\n\n")?;
+            }
+            first = false;
+
+            write!(writer, "const {name}: ?;")?;
+        }
+
+        let mut func_infos = self.iter_fn().collect::<Vec<_>>();
+        func_infos.sort_by(|a, b| a.metadata.name.cmp(&b.metadata.name));
+
+        for f in func_infos {
+            if !first {
+                writer.write_str("\n\n")?;
+            }
+            first = false;
+
+            if f.metadata.access == FnAccess::Private {
+                continue;
+            }
+
+            f.write_declaration(
+                writer,
+                decl.engine.custom_keywords.contains_key(&f.metadata.name)
+                    || (!f.metadata.name.contains('$')
+                        && !is_valid_function_name(&f.metadata.name)),
+            )?;
+        }
+
+        Ok(())
+    }
+}
+
+impl FuncInfo {
+    fn write_declaration(&self, writer: &mut dyn fmt::Write, operator: bool) -> fmt::Result {
+        for comment in &*self.metadata.comments {
+            writeln!(writer, "{comment}")?;
+        }
+
+        if operator {
+            writer.write_str("op ")?;
+        } else {
+            writer.write_str("fn ")?;
+        }
+
+        if let Some(name) = self.metadata.name.strip_prefix("get$") {
+            write!(writer, "get {name}(")?;
+        } else if let Some(name) = self.metadata.name.strip_prefix("set$") {
+            write!(writer, "set {name}(")?;
+        } else {
+            write!(writer, "{}(", self.metadata.name)?;
+        }
+
+        let mut first = true;
+        for i in 0..self.metadata.params {
+            if !first {
+                writer.write_str(", ")?;
+            }
+            first = false;
+
+            let (param_name, param_type) = self
+                .metadata
+                .params_info
+                .get(i)
+                .map(|s| {
+                    let mut s = s.split(':');
```

Comment:
You might want to do `s.splitn(':', 2);` here, otherwise:  `x : Range<path::to::my::type>` gets also split via the `::` path separators.

Reply:
Ah yeah that was it, thanks.